miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/StorageLocationRestController.java

```diff
@@ -0,0 +1,200 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response.Status;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StorageLocation;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StorageLocation.LocationUnit;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.StorageLocationDto;
+import uk.ac.bbsrc.tgac.miso.service.StorageLocationService;
+
+@Controller
+@RequestMapping("/rest/storagelocations")
+public class StorageLocationRestController extends RestController {
+
+  @Autowired
+  private StorageLocationService storageLocationService;
+
+  @RequestMapping(value = "/freezers", method = RequestMethod.GET)
+  public @ResponseBody List<StorageLocationDto> getFreezers() {
+    return storageLocationService.listFreezers().stream()
+        .map(freezer -> Dtos.asDto(freezer, false, false))
+        .collect(Collectors.toList());
+  }
+
+  @RequestMapping(value = "/bybarcode", method = RequestMethod.GET)
+  public @ResponseBody StorageLocationDto getLocation(@RequestParam(name = "q", required = true) String barcode) {
+    StorageLocation location = storageLocationService.getByBarcode(barcode);
+    if (location == null) {
+      throw new RestException("storage location not found", Status.NOT_FOUND);
+    }
+    return Dtos.asDto(location, true, false);
+  }
+
+  @RequestMapping(value = "/{id}/children", method = RequestMethod.GET)
+  public @ResponseBody List<StorageLocationDto> getChildLocations(@PathVariable(name = "id", required = true) long id) {
+    StorageLocation location = storageLocationService.get(id);
+    if (location == null) {
+      throw new RestException("storage location not found", Status.NOT_FOUND);
+    }
+    return location.getChildLocations().stream()
+        .map(child -> Dtos.asDto(child, false, false))
+        .collect(Collectors.toList());
+  }
+
+  @RequestMapping(value = "/freezers", method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.CREATED)
+  public @ResponseBody StorageLocationDto createFreezer(@RequestBody StorageLocationDto dto) {
+    return doSave(dto);
+  }
+
+  @RequestMapping(value = "/freezers/{id}", method = RequestMethod.PUT)
+  public @ResponseBody StorageLocationDto update(@PathVariable(name = "id", required = true) long id, @RequestBody StorageLocationDto dto)
+      throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)